### ISSUE TITLE
Server publish state should be dependent on deployables publish states

### ISSUE THREAD

**Issue by robstryker** [redhat-developer/rsp-server#211]
A server with some deployments should have the overall publish state of the server be calculated to some extent from the publish states of the modules below it. For example, if a synchronized server has a new deployment added (or an existing one remove), the publish state of the server should be "full". If a deployment has an 'unknown' state, then the state of the server should also be unknown. 

In short, the total publish state of a server is really an accumulation or worst-case of the publish state of the various deployments.

**Comment by adietish** [comment]
steps:

1. ASSERT: start with clean server, no deployments. 
1. EXEC: Add deployment.  
1. ASSERT: Make sure server publish state = "full".   
1. EXEC: call "publish".  
1. ASSERT: Verify server and deployments both have publish state "None"
1. EXEC: remove deployment. 
1. ASSERT: Verify server publish state is "Full" and module publish state is "Remove"
